### ISSUE TITLE
Clarify sentence prohibiting blank sections that also contain no Subsections in CPs and CPSes

### ISSUE THREAD

**Issue by BenWilson-Mozilla** [mozilla/pkipolicy#263]
In section 3.3, item 5, it says that CPs/CPSes cannot have subsections – “and have no subsections”. However, the Baseline Requirements and other RFC 3647-formatted documents have subsections. I think the intent of the existing language was to prevent empty subsections and not a strict prohibition of subsections that are not in RFC 3647.  I don't believe that RFC 3647 prohibits more granular outlining - it's a "framework".

**Comment by WilsonKathleen** [comment]
Here's more context:
"All CPs, CPSes, and combined CP/CPSes MUST be structured according to RFC 3647 and MUST: 
...
contain no sections that are blank and have no subsections; "

so perhaps it is just missing the word "both"; e.g.

contain no sections that are both blank and have no subsections;

**Comment by BenWilson-Mozilla** [comment]
According to Kathleen, the purpose is to prevent sections with no text at all.  It is OK for a section to have no text directly in it, as long as it has non-empty subsections.

**Comment by timfromdigicert** [comment]
Right.  I remember working with Kathleen on the wording for this, it seems like we didn't quite get it perfect.  I'd suggest:

"All CPs, CPSes, and combined CP/CPSes MUST be structured according to RFC 3647 and MUST:
...
contain no sections that are entirely blank, having no text or subsections; "

**Comment by timfromdigicert** [comment]
Another thing that has been pointed out to me by one of the RFC 3647 authors is that RFC 3647 is informative and has no requirements, so it's not precisely clear what "structured according to RFC 3647" actually means in practice.

It's not the world's most important problem, but most of the content Mozilla actually wants is already helpfully described in the first three paragraphs of RFC 3647, so perhaps "structured according to" would be better as "MUST follow the outline structure and recommendations described in RFC 3647 section 4."

This makes it clear what the requirement actually is (use the agreed upon outline, including no stip where appropriate),

**Comment by BenWilson-Mozilla** [comment]
Reopening because _**somehow**_ it does not appear that this change was merged into the final version 2.9. I'll have to investigate.

**Comment by BenWilson-Mozilla** [comment]
Apparently, the Mozilla GitHub repository was out of sync with my repository because we reverted a commit that addressed this issue. Rather than trying to fix this minor issue for publication in MRSP version 2.9, we have decided to include it in MRSP v. 3.0.

**Comment by BenWilson-Mozilla** [comment]
See discussion on the Mozilla dev-security-policy list, started here: https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/7ZyFP9Qr59o/m/A1sopH1RCwAJ

**Comment by BenWilson-Mozilla** [comment]
Section 3.3 of MRSP v.3.0 now states, "all CPs, CPSes, and combined CP/CPSes MUST be structured according to the common outline set forth in [section 6 of [RFC 3647](https://datatracker.ietf.org/doc/html/rfc3647#section-6), as may be amended by the CA/Browser Forum's TLS BRs or its S/MIME BRs, and MUST:
- include at least every section and subsection defined in [section 6 of RFC 3647](https://datatracker.ietf.org/doc/html/rfc3647#section-6);
- only use the words "No Stipulation" to mean that the particular document imposes no requirements related to that section; and
- contain no sections that are entirely blank, having no text or subsections;"
